include/alpaka/dev/DevUniformCudaHipRt.hpp

```diff
@@ -43,7 +44,30 @@ namespace alpaka
     {
         template<typename TApi, bool TBlocking>
         class QueueUniformCudaHipRt;
-    }
+
+        template<typename TApi>
+        class DevUniformCudaHipRtImpl
+        {
+        public:
+            DevUniformCudaHipRtImpl() = default;
+            DevUniformCudaHipRtImpl(int iDevice) : m_iDevice{iDevice}
+            {
+            }
+
+            [[nodiscard]] auto getNativeHandle() const noexcept -> int
+            {
+                return m_iDevice;
+            }
+            auto sendTaskToThePool(std::packaged_task<void()>&& task) -> std::future<void>
+            {
+                return callbackPool.submit(std::move(task));
+            }
+
+        private:
+            int m_iDevice;
+            core::ThreadPool callbackPool;
+        };
```

Comment:
As @fwyzard suggested, maybe just use a single threadpool for entire alpaka:

```suggestion
auto& getThreadPool() -> core::ThreadPool {
    static core::ThreadPool pool;
    return pool;
}
```

Reply:
Should I revert back to the Device only (without implementation) or do we want to keep the refactoring?

Reply:
What is the benefit of the `DevUniformCudaHipRtImpl` class? I would rather remove it, if there is no compelling reason to keep it.

Reply:
`auto&` with trailing return type (rather than plain `auto`) apparently makes the compiler angry. So I just went for `core::ThreadPool&` in the declaration

Reply:
Sorry, error on my side. Either `auto& getThreadPool() {...}` or `auto getThreadPool() -> core::ThreadPool& {...}` are correct. I prefer the latter.